lib/puppet/functions/run_task.rb

```diff
@@ -21,23 +21,46 @@ def mocked_run_task(task, *hosts)
         {:operation => 'run_task'})
     end
 
+    begin
+      require 'bolt/executor'
+      require 'bolt/node'
+    rescue LoadError
+      raise Puppet::ParseErrorWithIssue.from_issue_and_stack(Puppet::Pops::Issues::TASK_MISSING_BOLT)
+    end
```

Comment:
The above is best done with a feature - see how this is done in "eyaml_lookup_key.rb" 
```
  unless Puppet.features.hiera_eyaml?
    raise Puppet::DataBinding::LookupError, 'Lookup using eyaml lookup_key function is only supported when the hiera_eyaml library is present'
  end
```

Reply:
and that snippet is placed before the call to dispatch